spec/features/course/achievement_management_spec.rb

```diff
@@ -1,136 +1,241 @@
+          new_description = 'New description'
+          fill_in 'achievement_title', with: new_title
+          fill_in 'achievement_description', with: new_description
+          click_button I18n.t('helpers.submit.achievement.update')
+          expect(current_path).to eq course_achievements_path(course)
+          expect(page).to have_selector('div', I18n.t('course.achievements.update.success'))
+          expect(achievement.reload.title).to eq(new_title)
+          expect(achievement.reload.description).to eq(new_description)
         end
+      end
 
-        it 'shows errors' do
-          expect(page).to have_css('div.has-error')
+      describe 'I can create an achievement condition' do
+        scenario 'if there is a valid achievement which can serve as a condition' do
+          achievement = create(:course_achievement, course: course)
+          visit edit_course_achievement_path(course, achievement)
+          condition_achievement = create(:course_achievement, course: course)
+          expect(page).to have_selector(:link,
+                                        I18n.t('course.condition.achievements.new.header'))
+          click_link I18n.t('course.condition.achievements.new.header')
+          expect(current_path).
+            to eq(new_course_achievement_condition_achievement_path(course, achievement))
+          find('#condition_achievement_achievement_id').
+            find(:css, "option[value='#{condition_achievement.id}']").
+            select_option
+          click_button I18n.t('helpers.submit.condition_achievement.create')
+          expect(current_path).to eq edit_course_achievement_path(course, achievement)
+          expect(page).to have_selector('tr.condition > td:nth-child(2)',
+                                        text: condition_achievement.title)
         end
-      end
 
-      context 'with valid information' do
-        let(:new_title)  { 'New Title' }
-        let(:new_description) { 'New description' }
+        scenario 'unless there are no valid achievements' do
+          Course::Condition::Achievement.destroy_all
+          Course::Achievement.destroy_all
+          achievement = create(:course_achievement, course: course)
+          visit edit_course_achievement_path(course, achievement)
 
-        before do
-          fill_in 'achievement_title',        with: new_title
-          fill_in 'achievement_description',      with: new_description
-          click_button I18n.t('helpers.submit.achievement.update')
+          # The creation link won't be shown
+          expect(page).not_to have_selector(:link,
+                                            I18n.t('course.condition.achievements.new.header'))
         end
 
-        it 'redirects the user to index page' do
-          expect(current_path).to eq course_achievements_path(course)
+        scenario 'unless it has already been added or is the current one' do
+          achievement_condition, achievement, other_achievement =
```

Comment:
Useless assignment to variable - `achievement_condition`. Use `_` or `_achievement_condition` as a variable name to indicate that it won't be used.